record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java

```diff
@@ -64,6 +59,13 @@
         typeVariables = record.getTypeParameters().stream().map(TypeVariableName::get).collect(Collectors.toList());
         recordComponents = record.getRecordComponents().stream().map(ElementUtils::getClassType).collect(Collectors.toList());
         uniqueVarName = getUniqueVarName();
+        if (metaData.inheritComponentAnnotations()) {
+            accessorAnnotations = record.getRecordComponents().stream().map(e -> e.getAccessor().getAnnotationMirrors()).collect(Collectors.toList());
+            canonicalConstructorAnnotations = ElementUtils.findCanonicalConstructor(record).map(constructor -> ((ExecutableElement)constructor).getParameters().stream().map(Element::getAnnotationMirrors).collect(Collectors.toList())).orElse(List.of());
```

Comment:
I would find the following (without my comments explaining what I did, all starting with `// <-`) slightly easier to read; however functionally it should do the same.
```java
accessorAnnotations = record.getRecordComponents()
        .stream()
        .map(RecordComponentElement::getAccessor)
        .map(Element::getAnnotationMirrors) // <- Make the mappings more explicit
        .collect(toList()); // <- Static import of Collectors.toList()
canonicalConstructorAnnotations = ElementUtils.findCanonicalConstructor(record)
        .map(ExecutableElement.class::cast) // <- Move the casting up
        .map(ExecutableElement::getParameters) // <- Make the mappings more explicit
        .map(parameters -> parameters.stream()
                .map(Element::getAnnotationMirrors)
                .collect(toList()) // <- Static import of Collectors.toList()
        ).orElse(List.of());
```